src/gui/labeling/qgslabelinggui.cpp

```diff
@@ -898,7 +929,8 @@ void QgsLabelingGui::updateGeometryTypeBasedWidgets()
 
 void QgsLabelingGui::showGeometryGeneratorExpressionBuilder()
 {
-  QgsExpressionBuilderDialog expressionBuilder( mLayer );
+  QgsVectorLayer *vLayer = qobject_cast< QgsVectorLayer * >( mLayer );
+  QgsExpressionBuilderDialog expressionBuilder( vLayer );
```

Comment:
I'd throw a `if ( !vLayer ) return` safety check in here

Reply:
gentle ping @nyalldawson